src/main/java/com/ldtteam/structures/blueprints/v1/Blueprint.java

```diff
@@ -76,19 +81,54 @@
      * @param requiredMods the required mods.
      */
     protected Blueprint(
-      short sizeX, short sizeY, short sizeZ, short palleteSize, IBlockState[] pallete,
-      short[][][] structure, NBTTagCompound[] tileEntities, List<String> requiredMods)
+      short sizeX,
+      short sizeY,
+      short sizeZ,
+      short palleteSize,
+      List<IBlockState> pallete,
+      short[][][] structure,
+      NBTTagCompound[] tileEntities,
+      List<String> requiredMods)
     {
         this.sizeX = sizeX;
         this.sizeY = sizeY;
         this.sizeZ = sizeZ;
         this.palleteSize = palleteSize;
         this.palette = pallete;
         this.structure = structure;
-        this.tileEntities = tileEntities;
+        this.tileEntities = new NBTTagCompound[sizeY][sizeZ][sizeX];
+
+        for (final NBTTagCompound te : tileEntities)
+        {
+            if (te != null)
+            {
+                this.tileEntities[te.getShort("y")][te.getShort("z")][te.getShort("x")] = te;
+            }
+        }
         this.requiredMods = requiredMods;
     }
 
+    /**
+     * Constructor of a new Blueprint.
+     *
+     * @param sizeX        the x size.
+     * @param sizeY        the y size.
+     * @param sizeZ        the z size.
+     */
+    public Blueprint(short sizeX, short sizeY, short sizeZ)
+    {
+        this.sizeX = sizeX;
+        this.sizeY = sizeY;
+        this.sizeZ = sizeZ;
+        this.structure = new short[sizeY][sizeZ][sizeX];
+        this.tileEntities = new NBTTagCompound[sizeY][sizeZ][sizeX];
+        this.entities = new NBTTagCompound[sizeY][sizeZ][sizeX];
+
+        this.requiredMods = new ArrayList<>();
+        this.palette = new ArrayList<>();
```

Comment:
Missing types <>, but I will do a cleaning later..

Reply:
it doesn't need it here, it has it above.